src/pages/home/HomePage.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState, type FC } from 'react';
+import { Outlet } from 'react-router';
+import { Pagination } from '../../components/pagination/Pagination';
+import { SearchForm } from '../../components/search/SearchForm/SearchForm';
+import { useFilter } from '../../hooks/useFilter';
+import { Loader } from '../../components/loader/Loader';
+import { ResultsList } from './components/ResultsList/ResultsList';
+import { SelectedFlyout } from './components/SelectedFlyout/SelectedFlyout';
+
+export type Character = {
+  id: number;
+  name: string;
+  status: string;
+  species: string;
+  image: string;
+};
+
+export type ServerData = {
+  info: Record<string, unknown>;
+  results: Character[];
+};
+
+export const HomePage: FC = () => {
+  const [query, setQuery] = useState('');
+  const [serverData, setServerData] = useState<ServerData>({
+    info: {},
+    results: [],
+  });
+  const [totalPages, setTotalPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [isLoading, setIsLoading] = useState(false);
+  const renderData = useFilter(serverData, query);
+
+  const serverUrl = 'https://rickandmortyapi.com/api';
```

Comment:
Such thing should be a constant and  be placed to the `constant.ts` of `src`
`const SERVER_URL = 'https://rickandmortyapi.com/api'`